src/dataflow/components/dataflow-program.tsx

```diff
@@ -160,13 +164,43 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
         if (change) {
           (async () => {
             await engine.abort();
-            await engine.process(editor.toJSON());
+            await engine.process(this.programEditor.toJSON());
           })();
         }
       });
 
-      editor.view.resize();
-      (editor as any).trigger("process");
+      this.programEditor.view.resize();
+      (this.programEditor as any).trigger("process");
+
+      this.intervalHandle = setInterval(this.heartBeat.bind(this), HEARTBEAT_INTERVAL);
+
     })();
   }
+
+  public componentWillUnmount() {
+    clearInterval(this.intervalHandle);
+  }
+
+  private heartBeat() {
+    this.programEditor.nodes.forEach((n: Node) => {
```

Comment:
Do we need to worry about order of visiting nodes at all?

Reply:
Once we add the generators, we will at least need to make sure that we update the generator values and THEN update the plots.  So control/data order, but not necessarily node order.  Can't think of a reason yet where node order is important.